src/semgrep_mcp/semgrep.py

```diff
@@ -0,0 +1,215 @@
+import asyncio
+import json
+import os
+import subprocess
+from typing import Any
+
+from mcp.shared.exceptions import McpError
+from mcp.types import INTERNAL_ERROR, ErrorData
+
+################################################################################
+# Prelude #
+################################################################################
+# Communicating with the Semgrep binary.
+
+################################################################################
+# Constants #
+################################################################################
+
+_semgrep_lock = asyncio.Lock()
+
+# Global variable to store the semgrep executable path
+semgrep_executable: str | None = None
```

Comment:
What do you think about making this upper-case and keeping all globals like that?